### PR TITLE
Submission status update

### LINKED ISSUE #1419: [🐛 BUG] Submission status in Scenario Viewer not updating properly
### What went wrong? 🤔

The Submission Status is sometimes not appropriately aligned with the current jobs' statuses. For example, the status is "blocked" even if all the jobs have been completed.

Here is my scenario viewer:

image

Here are my jobs (all of them are completed):

image

### Expected Behavior

The status should properly reflect the status of the submission/jobs.

### Steps to Reproduce Issue

This happens when the scenario is more complex. I wasn't able to replicate it with a simple example. Here is some code to test it out.

https://github.com/FlorianJacta/TTT-SDM-Interview/tree/main/src_complete

Run main.py, create a scenario, and submit it. Then, you can see the status of the scenario viewer and the jobs' status inside the job selector.

### Version of Taipy

develop - 6/17/24

### Acceptance Criteria

- [ ] Ensure the new code is unit tested, and check that the code coverage is at least 90%.
- [ ] Create related issue in taipy-doc for documentation and Release Notes.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
resolves #1419

### CODE REVIEW

**Review by FabienLelaquais — DISMISSED** [review]
🤔whaaat?

File: frontend/taipy-gui/src/utils/index.ts [hunk 1]
```diff
@@ -105,14 +105,20 @@ export const getDateTimeString = (
...
+    const dateVal = getDateTime(value);
+    try {
+        if (withTime) {
+            return formatInTimeZone(
+                dateVal || "",
+                formatConf.forceTZ || !tz ? formatConf.timeZone : tz,
+                datetimeformat || formatConf.dateTime
+            );
+        }
+        return format(dateVal || 0, datetimeformat || formatConf.date);
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/utils/index.ts:117]: An easy thing that you could do first is simply to check if the format string (datetimeformat or formatConf.dateTime) contains a "D" or not. No need for fancy regexp here.
Add the useAdditionalDayOfYearTokens settings if yes, and we're all good.
What do you think?

**Review by FabienLelaquais — APPROVED** [review]
Great stuff.
A comment that I think is worth looking at on frontend/taipy-gui/src/utils/index.ts.

File: frontend/taipy-gui/src/utils/index.ts (hunk 1 above)
Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/utils/index.ts:117]: Then we may as well add the option all the time?

**Review by namnguyen20999 — APPROVED** [review]
lgtm !

**Pull request merged**